// rcnlx/src/projectFiles/question.js
module.exports.ban = [
  {
    title: "guildBanAdd",
    desc: "Emitted whenever a member is banned from a guild.",
  },
  {
    title: "guildBanRemove",
    desc: "Emitted whenever a member is unbanned from a guild.",
  },
];

module.exports.channel = [
  {
    title: "channelCreate",
    desc: "Emitted whenever a channel is created.",
  },
  {
    title: "channelDelete",
    desc: "Emitted whenever a channel is deleted.",
  },
  {
    title: "channelPinsUpdate",
    desc:
      "Emitted whenever the pins of a channel are updated. Due to the nature of the WebSocket event, not much information can be provided easily here - you need to manually check the pins yourself.",
  },
  {
    title: "channelUpdate",
    desc:
      "Emitted whenever a channel is updated - e.g. name change, topic change.",
  },
  {
    title: "typingStart",
    desc: "Emitted whenever a user starts typing in a channel.",
  },
  {
    title: "typingStop",
    desc: "Emitted whenever a user stops typing in a channel.",
  },
  {
    title: "voiceStateUpdate",
    desc:
      "Emitted whenever a user changes voice state - e.g. joins/leaves a channel, mutes/unmutes.",
  },
];

module.exports.emoji = [
  {
    title: "emojiCreate",
    desc: "Emitted whenever a custom emoji is created in a guild.",
  },
  {
    title: "emojiDelete",
    desc: "Emitted whenever a custom guild emoji is deleted.",
  },
  {
    title: "emojiUpdate",
    desc: "Emitted whenever a custom guild emoji is updated.",
  },
];

module.exports.guild = [
  {
    title: "guildCreate",
    desc: "Emitted whenever the client joins a guild.",
  },
  {
    title: "guildDelete",
    desc: "Emitted whenever a guild is deleted/left.",
  },
  {
    title: "guildUnavailable",
    desc:
      "Emitted whenever a guild becomes unavailable, likely due to a server outage.",
  },
  {
    title: "guildUpdate",
    desc: "Emitted whenever a guild is updated - e.g. name change.",
  },
];

module.exports.member = [
  {
    title: "guildMemberAdd",
    desc: "Emitted whenever a user joins a guild.",
  },
  {
    title: "guildMemberAvailable",
    desc: "Emitted whenever a member becomes available in a large guild.",
  },
  {
    title: "guildMemberRemove",
    desc: "Emitted whenever a member leaves a guild, or is kicked.",
  },
  {
    title: "guildMembersChunk",
    desc:
      "Emitted whenever a chunk of guild members is received (all members come from the same guild).",
  },
  {
    title: "guildMemberSpeaking",
    desc: "Emitted once a guild member starts/stops speaking.",
  },
  {
    title: "guildMemberUpdate",
    desc:
      "Emitted whenever a guild member changes - i.e. new role, removed role, nickname.",
  },
  {
    title: "presenceUpdate",
    desc:
      "Emitted whenever a guild member's presence changes, or they change one of their details.",
  },
  {
    title: "userUpdate",
    desc: "Emitted whenever a user's details (e.g. username) are changed.",
  },
];

module.exports.message = [
  {
    title: "message",
    desc: "Emitted whenever a message is created.",
  },
  {
    title: "messageDelete",
    desc: "Emitted whenever a message is deleted.",
  },
  {
    title: "messageDeleteBulk",
    desc: "Emitted whenever messages are deleted in bulk.",
  },
  {
    title: "messageReactionAdd",
    desc: "Emitted whenever a reaction is added to a message..",
  },
  {
    title: "messageReactionRemove",
    desc: "Emitted whenever a reaction is removed from a message.",
  },
  {
    title: "messageReactionRemoveAll",
    desc: "Emitted whenever all reactions are removed from a message.",
  },
  {
    title: "messageUpdate",
    desc:
      "Emitted whenever a message is updated - e.g. embed or content change.",
  },
];

module.exports.roles = [
    {
        title: "roleCreate",
        desc: "Emitted whenever a role is created.",
      },
      {
        title: "roleDelete",
        desc: "Emitted whenever a guild role is deleted.",
      },
      {
        title: "roleUpdate",
        desc: "Emitted whenever a guild role is updated.",
      }
]
